app/js/components/drugOrderEntry/addForm/AddForm.jsx

```diff
@@ -95,48 +107,58 @@ export class AddForm extends React.Component {
-        dispensingUnit,
-        dispensingQuantity,
-        reason,
-        previousOrder: this.state.previousOrder,
-        drugInstructions,
-        careSetting: this.props.careSetting.uuid,
-        drug: this.props.drugUuid,
-        orderer: this.props.session.currentProvider.uuid,
-        dosingType: this.state.formType === 'Standard Dosage' ?
-          'org.openmrs.SimpleDosingInstructions' :
-          'org.openmrs.FreeTextDosingInstructions',
-        type: "drugorder",
-        orderNumber: (this.state.action === 'NEW') ? this.props.draftOrders.length :
-          this.props.orderNumber,
-      },
-    }, () => {
-      this.props.addDraftOrder(this.state.draftOrder);
-      if (this.state.draftOrder.action === 'NEW') {
-        this.props.setOrderAction('DRAFT', this.state.orderNumber);
-      } else {
-        this.props.setOrderAction('DRAFT_EDIT', this.props.orderNumber);
-      }
-    });
-    this.props.selectDrugSuccess('');
-    this.clearDrugForms();
-    this.props.clearSearchField();
-    if (this.props.activity === 'DRAFT_ORDER_EDIT') {
-      this.props.setSelectedOrder({
-        currentOrderType: this.props.currentOrderType,
-        selectedOrder: null,
-        activity: null,
+    if (!this.checkIfDrugHasActiveOrder(this.props.drugName)) {
+      const {
+        previousOrder, formType, action, draftOrder,
+      } = this.state;
+      const {
+        careSetting, drugUuid, session, draftOrders, orderNumber, currentOrderType,
+      } = this.props;
+      this.setState({
+        draftOrder: {
+          drugName: this.props.drugName,
+          action: this.state.action,
+          dose,
+          dosingUnit,
+          frequency,
+          route,
+          duration,
+          durationUnit,
+          dispensingUnit,
+          dispensingQuantity,
+          reason,
+          previousOrder,
+          drugInstructions,
+          careSetting: careSetting.uuid,
+          drug: drugUuid,
+          orderer: session.currentProvider.uuid,
+          dosingType: formType === 'Standard Dosage' ?
+            'org.openmrs.SimpleDosingInstructions' :
+            'org.openmrs.FreeTextDosingInstructions',
+          type: "drugorder",
+          orderNumber: (action === 'NEW') ? draftOrders.length :
+            orderNumber,
+        },
+      }, () => {
+        this.props.addDraftOrder(draftOrder);
+        if (draftOrder.action === 'NEW') {
+          this.props.setOrderAction('DRAFT', this.state.orderNumber);
+        } else {
+          this.props.setOrderAction('DRAFT_EDIT', this.props.orderNumber);
+        }
       });
+      this.props.selectDrugSuccess('');
+      this.clearDrugForms();
+      this.props.clearSearchField();
+      if (this.props.activity === 'DRAFT_ORDER_EDIT') {
+        this.props.setSelectedOrder({
+          currentOrderType,
+          selectedOrder: null,
+          activity: null,
+        });
+      }
+    } else {
+      errorToast(`${this.props.drugName} Order is active.`);
```

Comment:
I think telling a user an order is active as a message might not be so direct. Maybe telling them that order already exists might be more descriptive. What do you think? Also, a small`o` would yield a better sentence. For example:`Paracetamol order already exists`.

Reply:
Will make the changes.

Reply:
Done